icu4c/source/test/hdrtst/testinternalheaders.sh

```diff
@@ -9,27 +9,38 @@
 CC=clang
 CXX=clang++
 
+ERROR_EXIT=0
+
 # Runtime libraries
 
 for file in `ls common/*.h`; do
     echo $file
     echo '#include "'$file'"' > ht_temp.cpp ;
     echo 'void noop() {}' >> ht_temp.cpp ;
     $CXX -c -std=c++11 -I common -DU_COMMON_IMPLEMENTATION -O0 ht_temp.cpp ;
+    if [ $? != 0 ] ; then
```

Comment:
Plausible. (I am no .sh expert. I assume you know this stuff or looked it up.)

Out of curiosity, could you achieve the same thing with replacing the semicolon on the previous line with `|| ERROR_EXIT=1`  ?

Reply:
> I think the most compact syntax would be
> $CXX -c -std=c++11 -I common -DU_COMMON_IMPLEMENTATION -O0 ht_temp.cpp || ERROR_EXIT=1

That's what I had in mind. I wasn't sure if you considered it and didn't like it.

> I can change it thus if you prefer.

I will leave it to you to decide which one is clearer/simpler/readable.

> As for if [ $? != 0 ] || ERROR_EXIT=1

That's not at all what I had in mind.

Reply:
I think many (most?) bash-like shells (bash, zsh, etc.) support the special variable `$?`.
Though perhaps one thought is that by using the pipe operator that might not depend on the particular shell implementation as much.

Reply:
I think $? As it is is more readable because it makes it clear that there's a control flow, not just an order of commands. $? is the common way to test the prior status, and had the advantage that you can check for a specific status not just nonzero.